Open Judge System/Workers/OJS.Workers.ExecutionStrategies/NodeJsZipPreprocessExecuteAndRunUnitTestsWithDOMAndMochaExecutionStrategy.cs

```diff
@@ -0,0 +1,322 @@
+
+            // Save code to file
+            var codeSavePath = FileHelpers.SaveStringToTempFile(codeToExecute);
+
+            // Create a Restricted Process Executor
+            IExecutor executor = new RestrictedProcessExecutor();
+
+            // Create a Checker using the information from the Execution Context
+            IChecker checker = Checker.CreateChecker(executionContext.CheckerAssemblyName, executionContext.CheckerTypeName, executionContext.CheckerParameter);
+
+            // Process tests 
+            result.TestResults = this.ProcessTests(executionContext, executor, checker, codeSavePath);
+
+            // Clean up
+            File.Delete(codeSavePath);
+
+            return result;
+        }
+
+        protected virtual string CreateSubmissionFile(ExecutionContext executionContext)
+        {
+            var trimmedAllowedFileExtensions = executionContext.AllowedFileExtensions?.Trim();
+
+            var allowedFileExtensions = (!trimmedAllowedFileExtensions?.StartsWith(".") ?? false)
+                ? $".{trimmedAllowedFileExtensions}"
+                : trimmedAllowedFileExtensions;
+
+            if (allowedFileExtensions != GlobalConstants.ZipFileExtension)
+            {
+                throw new ArgumentException("Submission file is not a zip file!");
+            }
+
+            return this.PrepareSubmissionFile(executionContext.FileContent);
+        }
+
+        protected virtual string PrepareSubmissionFile(byte[] submissionFileContent)
+        {
+            var submissionFilePath = $"{this.WorkingDirectory}\\{SubmissionFileName}";
+
+            File.WriteAllBytes(submissionFilePath, submissionFileContent);
+
+            this.ConvertContentToZip(submissionFilePath);
+
+            this.UnzipFile(submissionFilePath, this.WorkingDirectory);
+
+            return submissionFilePath;
+        }
+
+        protected virtual void ConvertContentToZip(string submissionZipFilePath)
+        {
+            using (var zipFile = new ZipFile(submissionZipFilePath))
+            {
+                zipFile.Save();
+            }
+        }
+
+        protected virtual void UnzipFile(string fileToUnzip, string outputDirectory)
+        {
+            using (var zipFile = ZipFile.Read(fileToUnzip))
+            {
+                foreach (var entry in zipFile)
+                {
+                    entry.Extract(outputDirectory, ExtractExistingFileAction.OverwriteSilently);
+                }
+            }
+
+            System.IO.File.Delete(fileToUnzip);
+        }
+
+        protected virtual string FindProgramEntryPath()
+        {
+            var files = new List<string>(Directory.GetFiles(this.WorkingDirectory, "app.js", SearchOption.AllDirectories));
+            if (files.Count == 0)
+            {
+                throw new ArgumentException($"'{AppJsFileName}' file not found in output directory!", nameof(this.WorkingDirectory));
```

Comment:
Please try not to go beyond 120 symbols in a row. -> move params of `ArgumentException` at new lines